common/src/main/java/io/quarkus/ts/openshift/common/Command.java

```diff
@@ -22,35 +28,67 @@ public Command(List<String> command) {
         this.command = command;
     }
 
+    public Command outputToFile(File file) {
+        outputConsumer = fileOutput(file);
+        return this;
+    }
+
+    public Command outputToConsole() {
+        outputConsumer = consoleOutput();
+        return this;
+    }
+
     private static String descriptionOfProgram(String program) {
         if (program.contains(File.separator)) {
             return program.substring(program.lastIndexOf(File.separator) + 1);
         }
         return program;
     }
 
-    public void runAndWait() throws IOException, InterruptedException {
+    public void runAndWait() {
         System.out.println(ansi().a("running ").fgYellow().a(String.join(" ", command)).reset());
+        try {
+            Process process = new ProcessBuilder()
+                    .redirectErrorStream(true)
+                    .command(command)
+                    .directory(new File(".").getAbsoluteFile())
+                    .start();
+
+            new Thread(() -> {
+                outputConsumer.accept(description, process.getInputStream());
+            }, "stdout consumer for command " + description).start();
 
-        Process process = new ProcessBuilder()
-                .redirectErrorStream(true)
-                .command(command)
-                .directory(new File(".").getAbsoluteFile())
-                .start();
+            int result = process.waitFor();
+            if (result != 0) {
+                throw new RuntimeException(description + " failed (executed " + command + ", return code " + result + ")");
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
```

Comment:
Any particular reason to wrap exceptions like this?  Whomever calls `runAndWait` should be prepared to deal with interruption.

Reply:
I will undo this and add the try/catch in the new callers.

Reply:
I think the callers can just propagate the exception, too. The generic "on failure action" interface should `throws Exception`, because it can execute arbitrary code.

That way, when error happens, you don't have a terribly long stack trace to go through.

Reply:
Not a big fan of having an interface with "throws Exception", I prefer specific exceptions instead. 
On the other hand, these interfaces are to be called when the test scenario fails. So, I guess it's fine to state that the method might throw any exception and this exception will not hide the original root cause of the test failure (it will only print a log message about the exception). 
So, I will change the interface definition and document it properly.

Reply:
Changed.

Reply:
I personally think that if the "on failure" action should be able to "do anything", then it naturally needs to be able to "throw anything". See e.g. `java.util.concurrent.Callable` :-)